state/core_access.go

```diff
@@ -177,6 +178,10 @@ func (ca *CoreAccessor) SubmitPayForBlob(
 		ca.lastPayForBlob = time.Now().UnixMilli()
 		ca.payForBlobCount++
 	}
+
+	if response.Code != 0 {
```

Comment:
very optional nit: can be coupled with the if above

Reply:
I would suggest keeping it separately: the previous statement contains metrics handling, and the new one is error handling.